packages/manager/src/features/Volumes/utils.ts

```diff
@@ -0,0 +1,78 @@
+import type { Event, Notification, Volume } from '@linode/api-v4';
+import type { Status } from 'src/components/StatusIcon/StatusIcon';
+
+export const volumeStatusIconMap: Record<Volume['status'], Status> = {
+  active: 'active',
+  creating: 'other',
+  migrating: 'other',
+  offline: 'inactive',
+  resizing: 'other',
+};
+
+/**
+ * Given an in-progress event and a volume's status, this function
+ * returns a volume's status with event info taken into account.
+ *
+ * We do this to provide users with a real-time feeling experience
+ * without having to refetch a volume's status agressivly.
+ *
+ * @param status The actual volume status from the volumes endpoint
+ * @param event An in-progress event for the volume
+ * @returns a volume status
+ */
+export const getDerivedVolumeStatusFromStatusAndEvent = (
+  status: Volume['status'],
+  event: Event | undefined
+): Volume['status'] => {
+  if (event === undefined) {
+    return status;
+  }
+
+  if (event.action === 'volume_migrate' && event.status === 'started') {
+    return 'migrating';
+  }
+
+  return status;
+};
+
+/**
+ * Returns a nicely formated percentage from an event
+ * only if the event is in progress and has a percentage.
+ *
+ * This allows us to show the user the progress of a
+ * volume migration.
+ *
+ * @returns "(50%)" for example
+ */
+export const getEventProgress = (event: Event | undefined) => {
+  if (
+    event === undefined ||
+    event.percent_complete === null ||
+    event.status !== 'started'
+  ) {
+    return null;
+  }
+
+  return `(${event.percent_complete}%)`;
+};
+
+export const getUpgradeableVolumeIds = (
```

Comment:
```suggestion
/**
 * Returns an array of IDs of Volumes that are scheduled to be upgraded.
 * 
 * @param volumes - Array from which to retrieve upgradeable Volumes.
 * @param notifications - Notifications containing Volume migration statuses.
 * 
 * @returns Array of upgradeable Volume IDs.
 */
export const getUpgradeableVolumeIds = (
```
Mind if we add doc comments for this util? (Not sure how accurate the suggestion above is)